src/shadowbox/server/manager_service.ts

```diff
@@ -192,6 +192,24 @@ function validateDataLimit(limit: unknown): DataLimit {
   return limit as DataLimit;
 }
 
+function validatePassword(password: string): string {
+  if (!password) {
+    throw new restifyErrors.MissingParameterError(
+      {statusCode: 400},
+      'Missing `password` parameter'
+    );
+  }
+
+  const bytesOfPassword = Buffer.byteLength(password as string, 'utf8');
+  if (bytesOfPassword < 22) {
```

Comment:
This kind of business logic should be in the repository instead, in the model layer (in terms of Domain-Driven Design). This file is just an application-layer adaptor to expose the domain objects, it should only do serialization/deserialization and type validation.

I'm also ok with removing the byte length check. If the service provider is using this option, they must know what they are doing, and the limit may get in the way.

Reply:
I still see this here. Let's just remove the length check.
We only need to check if it's a non empty string.

Reply:
The check for 22 bytes is still there. Please remove. We only need to check if it's a non empty string.